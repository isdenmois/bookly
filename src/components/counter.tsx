import React from 'react';
import { Text, View, ViewStyle } from 'react-native';
import { light, dark } from 'types/colors';
import { TextXL } from 'components/text';
import { DynamicStyleSheet, useDynamicValue, DynamicValue } from 'react-native-dynamic';

interface Props {
  value: string | number;
  label: string;
  testID?: string;
}

export function Counter(props: Props) {
  const s = useDynamicValue(ds);

  return (
    <View style={s.container}>
      <TextXL style={s.value} testID={props.testID}>
        {props.value}
      </TextXL>
      <Text style={s.label}>{props.label}</Text>
    </View>
  );
}

const ds = new DynamicStyleSheet({
  container: {
    flexDirection: 'column',
    justifyContent: 'center',
  } as ViewStyle,
  value: {
    color: new DynamicValue(light.PrimaryText, dark.PrimaryText),
    textAlign: 'center',
    marginBottom: 5,
    fontWeight: 'bold',
  },
  label: {
    color: new DynamicValue(light.PrimaryText, dark.PrimaryText),
    fontSize: 14,
  },
});
