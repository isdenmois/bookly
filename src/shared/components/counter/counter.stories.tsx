import React from 'react';
import { storiesOf } from '@storybook/react-native';
import { StyleSheet, View } from 'react-native';
import { Card } from 'components/card';
import { Counter } from './counter';

storiesOf('Counter')
  .add('Usage', () => (
    <View style={s.centered}>
      <Counter label='Прочитано' value='21' />
    </View>
  ))
  .add('Three counters', () => (
    <View style={[s.centered, s.threeContainer]}>
      <Card padding>
        <View style={s.threeRow}>
          <Counter label='Прочитано' value='19' />
          <Counter label='Запланировано' value='80' />
          <Counter label='Опережение' value='2' />
        </View>
      </Card>
    </View>
  ));

const s = StyleSheet.create({
  centered: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
  threeContainer: {
    paddingHorizontal: 24,
    alignItems: 'stretch',
  },
  threeRow: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
  },
});
