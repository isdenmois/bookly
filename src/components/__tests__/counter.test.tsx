import React from 'react';
import { create } from 'react-test-renderer';
import { Counter } from '../counter';

it('Counter', () => {
  const tree = create(<Counter value={10} label='Total' />);

  expect(tree.toJSON()).toMatchSnapshot();
});
