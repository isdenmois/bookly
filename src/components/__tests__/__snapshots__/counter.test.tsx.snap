// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter 1`] = `
<View
  style={
    Object {
      "flexDirection": "column",
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "fontSize": 24,
        },
        Object {
          "color": "#000",
          "fontWeight": "bold",
          "marginBottom": 5,
          "textAlign": "center",
        },
      ]
    }
  >
    10
  </Text>
  <Text
    style={
      Object {
        "color": "#000",
        "fontSize": 14,
      }
    }
  >
    Total
  </Text>
</View>
`;
